distribution/ltp/openposix_testsuite/patches/20140422/run-tests.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/sh
+#
+# A simple wrapper for pre- and post-execution activities for any given
+# openposix test.
+#
+# run_test contains logic moved out of Makefile.
+#
+# Garrett Cooper, June 2010
+#
+
+LOGFILE=${LOGFILE:=logfile}
+
+NUM_FAIL=0
+NUM_PASS=0
+NUM_TESTS=0
+
+run_test_loop() {
+
+	for t in $*; do
+
+		if run_test "$t"; then
+			NUM_PASS=$(expr $NUM_PASS + 1)
+		else
+			NUM_FAIL=$(expr $NUM_FAIL + 1)
+		fi
+		NUM_TESTS=$(expr $NUM_TESTS + 1)
+
+	done
+
+	cat <<EOF
+*******************
+SUMMARY
+*******************
+$(printf "PASS\t\t%3d" $NUM_PASS)
+$(printf "FAIL\t\t%3d" $NUM_FAIL)
+*******************
+$(printf "TOTAL\t\t%3d" $NUM_TESTS)
+*******************
+EOF
+
+}
+
+run_test() {
+
+	testname="$TEST_PATH/${1%.*}"
+
+	complog=$(basename $testname).log.$$
+
+	trace_cmd=""
+	if [ "$1" = "pthread_cond_broadcast_4-1.run-test" -o "$1" = "difftime_1-1.run-test" -o "$1" = "sigtimedwait_1-1.run-test" ]; then
+		trace_cmd="strace -tt -v -f"
+	fi
+	sh -c "$SCRIPT_DIR/t0 $TIMEOUT_VAL $trace_cmd ./$1 $(cat ./$(echo "$1" | sed 's,\.[^\.]*,,').args 2>/dev/null)" > $complog 2>&1
+
+	ret_code=$?
+
+	if [ "$ret_code" = "0" ]; then
+		echo "$testname: execution: PASS" >> "${LOGFILE}"
+		if [ -n "$trace_cmd" ]; then
+			cat $complog >> "${LOGFILE}"
+		fi
+	elif [ -f "$1" ]; then
+		case "$ret_code" in
+		1)
+			msg="FAILED"
+			;;
+		2)
+			msg="UNRESOLVED"
+			;;
+		4)
+			msg="UNSUPPORTED"
+			;;
+		5)
+			msg="UNTESTED"
+			;;
+		$TIMEOUT_RET)
+			msg="HUNG"
+			;;
+		*)
+			if [ $ret_code -gt 128 ]; then
+				msg="SIGNALED"
+			else
+				msg="EXITED ABNORMALLY"
+			fi
+		esac
+		echo "$testname: execution: $msg: Output: " >> "${LOGFILE}"
+		cat $complog >> "${LOGFILE}"
+		echo "$testname: execution: $msg "
+	else
+		echo "$testname: execution: SKIPPED (test not present)"
+	fi
+
+	rm -f $complog
+
+	return $ret_code
+
+}
+
+# SETUP
+if [ -w "$LOGFILE" ] || echo "" > "$LOGFILE"; then
+	:
+else
+	echo >&2 "ERROR: $LOGFILE not writable"
+	exit 1
```

Comment:
Same here, add an abort here if logfile isn't present ?

Reply:
Yes again, we should add an abort